virtual_rainforest/core/registry.py

```diff
@@ -67,156 +73,75 @@ def register_module(module_name: str) -> None:
 
     # Extract the last component of the module name to act as unique short name
     module_name_short = module_name.split(".")[-1]
-    is_not_core = module_name != "virtual_rainforest.core"
+    if module_name_short in MODULE_REGISTRY:
+        LOGGER.warning(f"Module already registered: {module_name_short}")
+        return
 
-    LOGGER.info(f"Registering {module_name} module components")
+    is_core = module_name == "virtual_rainforest.core"
 
-    # Create an entry for the module
-    if module_name_short not in MODULE_REGISTRY:
-        MODULE_REGISTRY[module_name_short] = {}
+    LOGGER.info(f"Registering {module_name} module components")
 
-    # Register the model unless this is core
-    if is_not_core:
-        register_model(
-            module=module, module_name=module_name, module_name_short=module_name_short
+    # Check on the model argument.
+    if is_core and model is not None:
+        msg = "No model should be registered for the core module."
+        LOGGER.critical(msg)
+        raise RuntimeError(msg)
+    elif not is_core and model is None:
+        msg = "A model class is required to register model modules."
+        LOGGER.critical(msg)
+        raise RuntimeError(msg)
+    elif model is not None:
+        if module_name_short != model.model_name:
+            msg = f"The model_name attribute and module name differ in {module_name}"
+            LOGGER.critical(msg)
+            raise RuntimeError(msg)
+
+        # Register the resulting single model class
+        LOGGER.info(
+            f"Registering model class for {module_name_short} model: {model.__name__}"
         )
 
     # Register the schema
-    schema_file_name = "model_schema.json" if is_not_core else "core_schema.json"
+    schema_file_name = "core_schema.json" if is_core else "model_schema.json"
```

Comment:
Is there a reason the core model's schema has a different file name to the others?

Reply:
No. I think we've already standardised  (e.g.) "plants_model_schema.json" to "model_schema.json". I don't want to have `core/model_schema.json` because it isn't a model schema. We could standardise everything to `module_schema.json` (or just `schema.json` 😄 ).

@jacobcook1995 ?

Reply:
I'll go ahead and standardise on `module_schema.json`